plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt

```diff
@@ -133,6 +146,80 @@ fun AndroidComponentsExtension<*, *, *>.configure(
     }
 }
 
+private fun Variant.configureDebugMetaPropertiesTask(
+    project: Project,
+    tasksGeneratingProperties: List<TaskProvider<*>>
+) {
+    if (isAGP74) {
+        project.logger.error("hello 74 ${AgpVersions.CURRENT}")
+        val taskSuffix = name.capitalized
+        val outputDir = project.layout.buildDirectory.dir(
+            "generated${sep}assets${sep}sentry${sep}debug-meta-properties$sep$name")
+        val generateDebugMetaPropertiesTask = SentryGenerateDebugMetaPropertiesTask.register(
+            project,
+            tasksGeneratingProperties,
+            outputDir,
+            taskSuffix
+        )
+
+        configureGeneratedSourcesFor74(
+            this,
+            generateDebugMetaPropertiesTask to DirectoryOutputTask::output
+        )
+    } else {
+        project.logger.error("not hello 74 ${AgpVersions.CURRENT}")
+        project.logger.info {
+            "Not configuring AndroidComponentsExtension for ${AgpVersions.CURRENT}, since it does" +
+                "not have new addGeneratedSourceDirectory API"
+        }
+    }
+}
+
+private fun Variant.configureSourceBundleTasks(project: Project, extension: SentryPluginExtension, cliExecutable: String): SourceContext.SourceContextTasks? {
+    if (extension.includeSourceBundle.get()) {
+        if (isAGP74) {
+            val androidAppExtension = project.extensions.getByType(AppExtension::class.java)
+            val paths = OutputPaths(project, name)
+            val taskSuffix = name.capitalized
+            val variant = AndroidVariant74(this)
+            val sourceFiles = androidAppExtension.sourceSets.flatMap {
```

Comment:
`h`: this is currently not considering `Variant`, so we'd always bundle all of the sources and not for the compiled variant. I think we should rather try to use `Variant` apis for that. I think you can get it via `variant.sources.java.all` or `variant.sources.kotlin.all`.

We could even extract this to the `AndroidVariant` interface and have different implementation for 7.0 and 7.4, but up to you.